fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -2363,8 +2396,16 @@ private List<Expr> getPredicatesBoundedByGroupbysSourceExpr(List<Expr> predicate
 
             boolean isAllSlotReferingGroupBys = true;
             for (SlotId slotId : slotIds) {
-                final SlotDescriptor slotDesc = analyzer.getDescTbl().getSlotDesc(slotId);
-                Expr sourceExpr = slotDesc.getSourceExprs().get(0);
+
+                Expr sourceExpr = new SlotRef(analyzer.getDescTbl().getSlotDesc(slotId));
+                while (sourceExpr instanceof SlotRef) {
```

Comment:
Why adding this unnest logic? Could you add an example in code to explain?

Reply:
add comment